WNPRC_Virology/src/org/labkey/wnprc_virology/notification/ViralLoadQueueNotification.java

```diff
@@ -0,0 +1,558 @@
+        }
+        return recipientName;
+    }
+
+    //override setuUp method?
+
+    //this setup method does most of the work to get the notify list
+    //if we want to change the location of where the notify list comes from we'll need to change this method
+    public void setUp()
+    {
+
+        // Set up query to get records for each rowid
+        Set<Integer> ids = new HashSet<Integer>();
+        Integer arr[] = new Integer[rowIds.length];
+        for (int i = 0; i < rowIds.length; i++){
+            arr[i] = Integer.parseInt(rowIds[i]);
+        }
+        Collections.addAll(ids, arr);
+        SimpleFilter filter = new SimpleFilter();
+        filter.addInClause(FieldKey.fromString("Key"), ids);
+        QueryHelper viralLoadQuery = new QueryHelper(container, currentUser, "lists", "vl_sample_queue");
+
+        // Define columns to get
+        List<FieldKey> columns = new ArrayList<>();
+        columns.add(FieldKey.fromString("Key"));
+        columns.add(FieldKey.fromString("CreatedBy"));
+        columns.add(FieldKey.fromString("Status"));
+        columns.add(FieldKey.fromString("Id"));
+        columns.add(FieldKey.fromString("emails"));
+        columns.add(FieldKey.fromString("ModifiedBy"));
+        columns.add(FieldKey.fromString("funding_string")); //account
+
+        String notifyEmails = null;
+
+        // Execute the query
+        try (Results rs = viralLoadQuery.select(columns, filter))
+        {
+            User mod = null;
+            while (rs.next()){
+                HashMap<String,Object> mp = new HashMap<>();
+                mp.put("Key", rs.getInt(FieldKey.fromString("Key")));
+                mp.put("emails", rs.getString(FieldKey.fromString("emails")));
+                mp.put("funding_string", rs.getInt(FieldKey.fromString("funding_string")));
+
+                VLSampleListResults.add(mp);
+
+                notifyEmails = rs.getString(FieldKey.fromString("emails"));
+
+                if (notifyEmails != null)
+                {
+                    this.notifyEmails = notifyEmails;
+                }
+                // old notes:
+                //emails.put(createdByUserId, getEmailArray(notifyEmails)); - old way, user id is unique.
+                //used to use public Map<Integer, List<String>> emails = new HashMap<Integer,List<String>>();
+                //but ideally instead of Integer as the key it would be a a string of:
+                //submitter email + notify email string(normalized = sorted in such a way there arent repeats).
+                countEmailsAndPut(notifyEmails);
+            }
+        }
+        catch (Exception e)
+        {
+            _log.error(this.getClass().getName() + ": issue getting sample queue list ", e);
+        }
+
+    }
+
+    public void getRowsAndSendMessage()
+    {
+        HashMap<String, Object> subscriberEmailSummary = new HashMap<>();
+        // was hoping to pull in grant account info since this is gathered in the trigger script on init,
```

Comment:
is this a future improvement to this method, and that's why you are keeping the comment?

Reply:
yeah if I can pull in the data that was already queried from the trigger script, it would save an extra database call, but the rhino engine that translates JS to Java didn't allow me to do this, so I left a note on what I tried.

Reply:
sounds good